### PR TITLE
feat: Toasts redesign

### PR DESCRIPTION
**Pull request by istarkov**
## Description

Based on https://www.figma.com/design/sfCE7iLS0k25qCxiifQNLE/%F0%9F%93%9A-Webstudio-Library?node-id=2-4053&m=dev

image

image

## Steps for reproduction

Open toast in storybook https://6382151c8b47d4399fb9fc69-qhzlpxqpli.chromatic.com/?path=/story/design-system-library-toast--in-action, 
or somehow cause error  or

run in dev console
```
window.__webstudio__$__builderApi.toast.success("dsdnsdnsmnd,smndm,sndm,snd sdm sdnsmndm,sndms d,m dsd sd d sds dsdsd sdsd sdsd s dsd s sdds", { icon: "🙂" });

window.__webstudio__$__builderApi.toast.success("dsdnsdnsmnd,smndm,sndm,snd sdm sdnsmndm,sndms d,m dsd sd d sds ds", { icon: "🙂" });
```

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env` file

### CODE REVIEW

File: packages/design-system/src/components/toast.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        </Toast>
+
+        <Toast icon={<ImageIcon />}>Asset already exists</Toast>
+      </StoryGrid>
+    </StorySection>
+  );
+};
+Design.storyName = "Toast Design";
+
+let i = 0;
```
Comment (kof) [inline packages/design-system/src/components/toast.stories.tsx:92]:
```suggestion
let index = 0;
```

Comment (kof) [reply packages/design-system/src/components/toast.stories.tsx:92]: or counter

File: packages/design-system/src/components/toast.tsx [hunk 2]
```diff
@@ -1,19 +1,28 @@
...
 import { Box } from "./box";
 import { theme } from "../stitches.config";
+import { Grid } from "./grid";
+import { AlertCircleIcon } from "@webstudio-is/icons";
+import { Text } from "./text";
+import { Tooltip } from "./tooltip";
+import { useState } from "react";
 
-const VIEWPORT_PADDING = theme.spacing[2];
+const VIEWPORT_PADDING = 8;
```
Comment (kof) [inline packages/design-system/src/components/toast.tsx:17]: if you need the number, use rawTheme

File: packages/design-system/src/components/toast.tsx [hunk 3]
```diff
@@ -71,67 +66,177 @@ const StyledToast = styled(ToastPrimitive.Root, {
...
       },
-      custom: {},
-      loading: {},
     },
   },
 });
 
+type ToastVariant = React.ComponentProps<typeof ToastVariants>["variant"];
+
+const cssVar = (name: string) => `var(${name})`;
```
Comment (kof) [inline packages/design-system/src/components/toast.tsx:103]: I know this is convenient, but I would prefer to either inline it or to have a single utility that is reusable, simply foc sonsistency

File: packages/design-system/src/components/toast.tsx [hunk 4]
```diff
@@ -71,67 +66,177 @@ const StyledToast = styled(ToastPrimitive.Root, {
...
+type Options = Pick<ToastOptions, "duration" | "id" | "icon">;
+
 export const toast = {
-  info: hotToast,
-  error: hotToast.error,
-  success: hotToast.success,
-  warn: hotToast.custom,
+  info: (value: string, options?: Options) => hotToast.success(value, options),
+  error: (value: string, options?: Options) => hotToast.error(value, options),
+  warn: (value: string, options?: Options) => hotToast.custom(value, options),
```
Comment (kof) [inline packages/design-system/src/components/toast.tsx:241]: why success gone?

**Review by kof — APPROVED** [review]

**Comment by kof** [comment]
- [ ] With enough content, it can grow too big, lets set some max height

image

**Comment by kof** [comment]
What we discussed

- [ ] 3 toast types: info, warning, error (gray, yellow, red)
- [ ] click on toast closes toast right away or alternetively starts a very fast animation (200ms) but animation starts RIGHT away, not in a second, not in two
- [ ] copy to buffer needs a copy icon-button

File: packages/design-system/src/components/toast.tsx (hunk 3 above)
Comment (istarkov) [reply packages/design-system/src/components/toast.tsx:105]: To early for the utility. Just played with the idea of using variables for all variants. I don't think that everything we are playing with needs an utility

**Review by kof — APPROVED** [review]
perfect!

**Pull request merged**